src/dune_rules/pkg_rules.ml

```diff
@@ -1799,28 +1829,49 @@ let setup_package_rules context ~dir ~pkg_name : Gen_rules.result Memo.t =
 
 let setup_rules ~components ~dir ctx =
   match components with
+  | [ ".dev-tool"; pkg_name; pkg_dep_name ] ->
```

Comment:
These ".dev-tool" are copy pasted far too many times.

Reply:
We could go for this at least to start experimenting the feature. And after refactor/improve later with your suggestion.

> To make the rules agnostic to "dev tools" simplest would be to allow for more than one lock directory to co-exist in a context. In the library database, every database can have a parent. Perhaps something like that can be done here as well?

I'm willing to land another PR.

Reply:
I've consolidated the separate logic for project dependencies and dev tools into the `Package_universe` module, so ".dev-tool" is now only repeated in this function.